feat: add OCI plugin distribution docs (#7245)
Kong/docs.konghq.com: app/_src/gateway/plugin-development/distribution.md

---

> title: Managing your custom plugin

Reviewer: Ditto on my comment in the TOC. IMO "Managing" is too generic.

> title: Managing your custom plugin

Reviewer 2: ```suggestion
title: Installation and Distribution
```

Kong Plugins can be packaged as generic OCI artefacts and uploaded to one of these 
registries for versioning, storage, and distribution. 

The advantage of distributing plugins as OCI artefacts is that users can make use of 
a number of ecosystem benefits including tooling around building, pushing/pulling, and 
signing (for secure provenance attestation) of these artefacts. The steps below 
illustrate a sample flow for packaging, distributing, and verifying a Kong custom plugin 
as an OCI artefact.

On the machine where the plugin is developed, or as part of an automated workflow, run the following 
steps:

1. Package the plugin according to the [Packaging Sources](#packaging-sources) section above.

    ```
    tar czf my-plugin.tar.gz ./my-plugin-dir
    ```

2. Use the OSS [Cosign tool][cosign-install] to generate a key pair for use signing and verifying plugins:

    ```
    cosign generate-key-pair
    ```

    The private key (`cosign.key`) should be kept secure and is used for signing the plugin artefact. The public key 
    (`cosign.pub`) should be distributed and used by target machines to validate the downloaded plugin later in the flow. 

>     There are also keyless methods for signing and verifying artefacts with Cosign. More information 

Reviewer 2: ```suggestion
    There are also key-less methods for signing and verifying artifacts with Cosign. More information 
```

    is available in their [documentation][cosign-signing].

> 3. Login to your OCI-compliant registry. In this case we'll use Dockerhub:

Reviewer 2: ```suggestion
3. Login to your OCI-compliant registry. In this case we'll use Docker Hub:
```

    ```
    cat ~/foo_password.txt | docker login --username foo-user --password-stdin
    ```

> 4. Upload the plugin artefact to the OCI registry using Cosign. This is the equivalent of running 

Reviewer 2: ```suggestion
4. Upload the plugin artifact to the OCI registry using Cosign. This is the equivalent of running 
```

> a `docker push <image>` when pushing a local Docker image up to a registry.

Reviewer 2: ```suggestion
`docker push <image>` when pushing a local Docker image up to a registry.
```
